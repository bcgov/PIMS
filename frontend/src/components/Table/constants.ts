export const DEFAULT_PAGE_SIZE = 10;
export const DEFAULT_PAGE_SELECTOR_OPTIONS: number[] = [5, 10, 20, 50, 100];
