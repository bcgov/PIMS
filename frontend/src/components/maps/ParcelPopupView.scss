@import "../../fonts.scss";
.parcelPopup {
  min-width: 211px;
  max-width: 250px;
  box-shadow: 1px 2px 2px;
  text-align: left;
  padding: 12px;
  .row {
    flex-direction: column;
  }
  .row .col :first-child {
    margin-top: 0px;
  }

  .label {
    font-weight: 700;
  }

  .list-group {
    margin-top: 6px;
  }
  .list-group-item {
    padding: 0;
    border: none;
    font-family: "BCSans" Fallback, sans-serif;
    font-size: 12px;
  }
  p {
      margin: 0;
  }
  .alert {
    margin: 0;
  }
}
