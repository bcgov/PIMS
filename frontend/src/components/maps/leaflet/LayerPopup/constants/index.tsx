import * as React from 'react';

export const MUNICIPALITY_LAYER_URL =
  'https://openmaps.gov.bc.ca/geo/pub/WHSE_LEGAL_ADMIN_BOUNDARIES.ABMS_MUNICIPALITIES_SP/wfs?SERVICE=WFS&REQUEST=GetFeature&VERSION=1.3.0&outputFormat=application/json&typeNames=pub:WHSE_LEGAL_ADMIN_BOUNDARIES.ABMS_MUNICIPALITIES_SP';
export const PARCELS_LAYER_URL =
  'https://openmaps.gov.bc.ca/geo/pub/WHSE_CADASTRE.PMBC_PARCEL_FABRIC_POLY_SVW/wfs?service=WFS&REQUEST=GetFeature&VERSION=1.3.0&outputFormat=application/json&typeNames=pub:WHSE_CADASTRE.PMBC_PARCEL_FABRIC_POLY_SVW';

export const parcelLayerPopupConfig = {
  PARCEL_NAME: { label: 'Parcel Name:', display: (data: any) => data.PARCEL_NAME },
  PIN: { label: 'Parcel PIN:', display: (data: any) => data.PIN },
  PID: { label: 'Parcel PID:', display: (data: any) => data.PID },
  PLAN_NUMBER: { label: 'Plan Number:', display: (data: any) => data.PLAN_NUMBER },
  PARCEL_CLASS: { label: 'Parcel Class:', display: (data: any) => data.PARCEL_CLASS },
  MUNICIPALITY: { label: 'Municipality:', display: (data: any) => data.MUNICIPALITY },
  REGIONAL_DISTRICT: {
    label: 'Regional District:',
    display: (data: any) => data.REGIONAL_DISTRICT,
  },
  FEATURE_AREA_SQM: {
    label: 'Area:',
    display: (data: any) => (
      <>
        {data.FEATURE_AREA_SQM} m<sup>2</sup>
      </>
    ),
  },
};

export const municipalityLayerPopupConfig = {
  ADMIN_AREA_GROUP_NAME: {
    label: 'Administration Area:',
    display: (data: any) => `${data.ADMIN_AREA_GROUP_NAME} (${data.CHANGE_REQUESTED_ORG})`,
  },
};
