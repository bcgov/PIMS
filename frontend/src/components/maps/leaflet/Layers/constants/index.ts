export const MUNICIPALITY_LAYER_URL =
  'https://openmaps.gov.bc.ca/geo/pub/WHSE_LEGAL_ADMIN_BOUNDARIES.ABMS_MUNICIPALITIES_SP/wfs?SERVICE=WFS&REQUEST=GetFeature&VERSION=1.3.0&outputFormat=application/json&typeNames=pub:WHSE_LEGAL_ADMIN_BOUNDARIES.ABMS_MUNICIPALITIES_SP';
export const PARCELS_LAYER_URL =
  'https://openmaps.gov.bc.ca/geo/pub/WHSE_CADASTRE.PMBC_PARCEL_FABRIC_POLY_SVW/wfs?service=WFS&REQUEST=GetFeature&VERSION=1.3.0&outputFormat=application/json&typeNames=pub:WHSE_CADASTRE.PMBC_PARCEL_FABRIC_POLY_SVW';

export const parcelLayerPopupConfig = {
  PIN: { label: 'Parcel PIN:', display: (data: any) => data.PIN },
  PID: { label: 'Parcel PID:', display: (data: any) => data.PID },
  REGIONAL_DISTRICT: {
    label: 'Regional District:',
    display: (data: any) => data.REGIONAL_DISTRICT,
  },
};

export const municipalityLayerPopupConfig = {
  ADMIN_AREA_NAME: { label: 'Area Name:', display: (data: any) => data.ADMIN_AREA_NAME },
  ADMIN_AREA_ABBREVIATION: {
    label: 'Area Abbrevation:',
    display: (data: any) => data.ADMIN_AREA_ABBREVIATION,
  },
  ADMIN_AREA_GROUP_NAME: {
    label: 'Area Group Name:',
    display: (data: any) => `${data.ADMIN_AREA_GROUP_NAME} (${data.CHANGE_REQUESTED_ORG})`,
  },
};
