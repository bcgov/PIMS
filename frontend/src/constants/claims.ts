/**
 * Claims enum, provides a list of permissions that govern what actions are available to an authenticated user.
 */
export enum Claims {
  PROPERTY_VIEW = 'property-view',
  PROPERTY_EDIT = 'property-edit',
  PROPERTY_ADD = 'property-add',
  PROPERTY_DELETE = 'property-delete',
  DISPOSE_REQUEST = 'dispose-request',
  DISPOSE_APPROVE = 'dispose-approve',
  ADMIN_USERS = 'admin-users',
  ADMIN_ROLES = 'admin-roles',
  ADMIN_PROPERTIES = 'admin-properties',
  ADMIN_PROJECTS = 'admin-projects',
  PROJECT_VIEW = 'project-view',
  PROJECT_EDIT = 'project-edit',
  PROJECT_ADD = 'project-add',
  PROJECT_DELETE = 'project-delete',
  REPORTS_VIEW = 'reports-view',
  REPORTS_SPL = 'reports-spl',
}

export default Claims;
