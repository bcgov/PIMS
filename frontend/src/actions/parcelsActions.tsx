import * as ActionTypes from 'constants/actionTypes';

//Parcel List API action

export interface IProperty {
  id: number;
  propertyTypeId: 0 | 1; // 0 = Parcel, 1 = Building
  latitude: number;
  longitude: number;
}

export interface IStoreParcelsAction {
  type: typeof ActionTypes.STORE_PARCEL_RESULTS;
  parcelList: IProperty[];
}

export const storeParcelsAction = (parcelList: IProperty[]): IStoreParcelsAction => ({
  type: ActionTypes.STORE_PARCEL_RESULTS,
  parcelList: parcelList,
});

//Parcel Detail API action
export interface IAddress {
  line1: string;
  line2?: string;
  city?: string;
  cityId: number | any;
  province?: string;
  provinceId: string;
  postal: string;
}

export interface IBuilding {
  id: number;
  localId: string;
  description: string;
  address: IAddress;
  latitude: number | any;
  longitude: number | any;
  buildingFloorCount: number | any;
  buildingConstructionType: string | undefined;
  buildingConstructionTypeId: number | any;
  buildingPredominateUse: string | undefined;
  buildingPredominateUseId: string | any;
  buildingOccupantType: string | undefined;
  buildingOccupantTypeId: number | any;
  leaseExpiry?: string;
  occupantName: string;
  transferLeaseOnSale: boolean;
  buildingTenancy: string;
  buildingNetBookValue: number;
  rentableArea: number | any;
  agencyId: number | any;
  evaluations: IEvaluation[];
}

export interface IEvaluation {
  fiscalYear: number | any;
  estimatedValue: number | any;
  assessedValue: number | any;
  netBookValue: number | any;
  appraisedValue: number | any;
}

export interface IParcel {
  id: number | any;
  pid: string;
  pin: string;
  latitude: number | any;
  longitude: number | any;
  statusId: number | any;
  propertyStatus: string | undefined;
  classification: string | undefined;
  classificationId: number | any;
  description: string;
  address: IAddress | any;
  landArea: number | any;
  landLegalDescription: string;
  zoning: string;
  zoningPotential: string;
  agency: string | undefined;
  agencyId: number | any;
  isSensitive: boolean;
  buildings: IBuilding[];
  evaluations: IEvaluation[];
}

export interface IAddress {
  line1: string;
  cityId: number | any;
  provinceId: string;
  postal: string;
}

export interface IParcelDetail {
  propertyTypeId: 0;
  parcelDetail: IParcel | null;
}

export interface IBuildingDetail {
  propertyTypeId: 1;
  parcelDetail: IBuilding | null;
}

export type IPropertyDetail = IParcelDetail | IBuildingDetail;

export interface IStoreParcelDetail {
  type: typeof ActionTypes.STORE_PARCEL_DETAIL;
  parcelDetail: IParcelDetail;
}

export const storeParcelDetail = (parcel: IParcel | null): IStoreParcelDetail => ({
  type: ActionTypes.STORE_PARCEL_DETAIL,
  parcelDetail: {
    propertyTypeId: 0,
    parcelDetail: parcel,
  },
});

export interface IStoreBuildingDetail {
  type: typeof ActionTypes.STORE_BUILDING_DETAIL;
  parcelDetail: IBuildingDetail;
}

export const storeBuildingDetail = (building: IBuilding | null): IStoreBuildingDetail => ({
  type: ActionTypes.STORE_BUILDING_DETAIL,
  parcelDetail: {
    propertyTypeId: 1,
    parcelDetail: building,
  },
});
