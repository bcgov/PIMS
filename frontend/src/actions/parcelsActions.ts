import * as ActionTypes from 'constants/actionTypes';
import { PointFeature } from 'components/maps/types';
import { PropertyTypes } from 'constants/propertyTypes';

//Parcel List API action

export interface IProperty {
  id: number | '';
  propertyTypeId?: PropertyTypes;
  agencyId?: number | '';
  latitude: number | '';
  longitude: number | '';
  name?: string;
  description?: string;
  projectNumbers?: string[];
  projectStatus?: string;
  isSensitive: boolean | '';
  createdOn?: string;
  updatedOn?: string;
  updatedByEmail?: string;
  updatedByName?: string;
}

export interface IStoreParcelsAction {
  type: typeof ActionTypes.STORE_PARCEL_RESULTS;
  parcelList: IProperty[];
}

export interface IStoreDraftParcelsAction {
  type: typeof ActionTypes.STORE_DRAFT_PARCEL_RESULTS;
  draftParcelList: PointFeature[];
}

export interface IStoreParcelAction {
  type: typeof ActionTypes.STORE_PARCEL_FROM_MAP_EVENT;
  parcel: IProperty;
}

export const storeDraftParcelsAction = (parcelList: PointFeature[]): IStoreDraftParcelsAction => ({
  type: ActionTypes.STORE_DRAFT_PARCEL_RESULTS,
  draftParcelList: parcelList,
});

export const storeParcelsAction = (parcelList: IProperty[]): IStoreParcelsAction => ({
  type: ActionTypes.STORE_PARCEL_RESULTS,
  parcelList: parcelList,
});

export const storeParcelAction = (parcel: IProperty): IStoreParcelAction => ({
  type: ActionTypes.STORE_PARCEL_FROM_MAP_EVENT,
  parcel,
});

//Parcel Detail API action
export interface IAddress {
  id?: number | undefined;
  line1: string;
  line2?: string;
  administrativeArea: string;
  province?: string;
  provinceId: string;
  postal: string;
}

export enum LeasedLand {
  owned = 0,
  leased = 1,
  other = 2,
}

export interface IBuilding extends IProperty {
  parcelId: number | '';
  pid: number | '';
  address: IAddress;
  buildingFloorCount?: number | '';
  buildingConstructionType?: string;
  buildingConstructionTypeId: number | '';
  buildingPredominateUse?: string;
  buildingPredominateUseId: number | '';
  buildingOccupantType?: string;
  buildingOccupantTypeId: number | '';
  classificationId: number | '';
  classification: string;
  encumbranceReason: string;
  leaseExpiry?: string;
  occupantName: string;
  transferLeaseOnSale: boolean;
  buildingTenancy: string;
  buildingTenancyUpdatedOn?: string;
  rentableArea: number | '';
  totalArea: number | '';
  agencyId: number | '';
  agency: string;
  agencyCode: string;
  subAgency?: string;
  assessedLand: number | '';
  assessedBuilding: number | '';
  evaluations: IEvaluation[];
  fiscals: IFiscal[];
  parcels: IParcel[];
}

export interface IFlatBuilding extends IProperty {
  parcelId: number;
  address: string;
  administrativeArea: string;
  postal: string;
  province: string;
  buildingFloorCount?: number | '';
  buildingConstructionType?: string;
  buildingConstructionTypeId: number | '';
  buildingPredominateUse?: string;
  buildingPredominateUseId: number | '';
  buildingOccupantType?: string;
  buildingOccupantTypeId: number | '';
  classificationId: number | '';
  classification: string;
  leaseExpiry?: string;
  occupantName: string;
  transferLeaseOnSale: boolean;
  buildingTenancy: string;
  rentableArea: number | '';
  agencyId: number | '';
  agency: string;
  agencyCode: string;
  assessedLand: number | '';
  assessedBuilding: number | '';
  netBook: number | '';
  leasedLand: {
    type: LeasedLand;
  };
}

export interface IFiscal {
  parcelId?: number;
  buildingId?: number;
  fiscalYear?: number | '';
  key: string;
  value: number | '';
  rowVersion?: string;
}

export interface IEvaluation {
  parcelId?: number;
  buildingId?: number;
  date?: Date | string;
  key: string;
  firm?: string;
  value: number | '';
  rowVersion?: string;
}

export interface IParcel extends IProperty {
  pid?: string;
  pin?: number | '';
  classification?: string;
  classificationId: number | '';
  encumbranceReason: string;
  address?: IAddress;
  landArea: number | '';
  landLegalDescription: string;
  zoning: string;
  zoningPotential: string;
  agency?: string;
  subAgency?: string;
  agencyId: number | '';
  buildings: IBuilding[];
  parcels: Partial<IParcel[]>;
  assessedLand: number | '';
  assessedBuilding: number | '';
  evaluations: IEvaluation[];
  fiscals: IFiscal[];
  rowVersion?: string;
}

export interface IFlatParcel extends IProperty {
  pid?: string;
  pin?: number | '';
  classification?: string;
  classificationId: number | '';
  address: string;
  administrativeArea: string;
  postal: string;
  landArea: number | '';
  landLegalDescription: string;
  zoning: string;
  zoningPotential: string;
  agency?: string;
  agencyId: number | '';
  isSensitive: boolean;
  buildings: IBuilding[];
  assessedLand: number | '';
  assessedBuilding: number | '';
  netBook: number | '';
}

export interface IParcelDetail {
  propertyTypeId: PropertyTypes;
  parcelDetail: IParcel | null;
  position?: [number, number]; // (optional) a way to override the positioning of the map popup
}

export interface IBuildingDetail {
  propertyTypeId: 1;
  parcelDetail: IBuilding | null;
  position?: [number, number]; // (optional) a way to override the positioning of the map popup
}

export type IPropertyDetail = IParcelDetail | IBuildingDetail;

export interface IStoreParcelDetail {
  type: typeof ActionTypes.STORE_PARCEL_DETAIL;
  parcelDetail: IParcelDetail;
}

export const storeParcelDetail = (
  parcel: IParcel | null,
  position?: [number, number],
): IStoreParcelDetail => {
  return {
    type: ActionTypes.STORE_PARCEL_DETAIL,
    parcelDetail: {
      propertyTypeId: parcel?.propertyTypeId as PropertyTypes,
      parcelDetail: parcel,
      position,
    },
  };
};

export interface IStoreBuildingDetail {
  type: typeof ActionTypes.STORE_BUILDING_DETAIL;
  parcelDetail: IBuildingDetail;
}

export const storeBuildingDetail = (
  building: IBuilding | null,
  position?: [number, number],
): IStoreBuildingDetail => ({
  type: ActionTypes.STORE_BUILDING_DETAIL,
  parcelDetail: {
    propertyTypeId: 1,
    parcelDetail: building,
    position,
  },
});

export interface IStoreAssociatedBuildingDetail {
  type: typeof ActionTypes.STORE_ASSOCIATED_BUILDING_DETAIL;
  associatedBuildingDetail: IBuildingDetail;
}

export const storeAssociatedBuilding = (
  building: IBuilding | null,
  position?: [number, number],
): IStoreAssociatedBuildingDetail => ({
  type: ActionTypes.STORE_ASSOCIATED_BUILDING_DETAIL,
  associatedBuildingDetail: {
    propertyTypeId: 1,
    parcelDetail: building,
    position,
  },
});
