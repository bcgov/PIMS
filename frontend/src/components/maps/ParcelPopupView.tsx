import * as React from 'react';
import { ListGroup, Container } from 'react-bootstrap';
import { IParcel } from 'actions/parcelsActions';
import { Alert, Row, Col } from 'react-bootstrap';
import { Label } from 'components/common/Label';
import './ParcelPopupView.scss';
import { Link } from 'react-router-dom';
import { EvaluationKeys } from '../../constants/evaluationKeys';
import useKeycloakWrapper from 'hooks/useKeycloakWrapper';

export interface IParcelDetailProps {
  parcel: IParcel | null;
  disabled?: boolean;
}

export const ParcelPopupView = (props: IParcelDetailProps | null) => {
  const parcelDetail: IParcel | null | undefined = props?.parcel;
  const keycloak = useKeycloakWrapper();

  return (
    <Container className="parcelPopup" fluid={true}>
      {!parcelDetail ? (
        <Alert variant="danger">Failed to load parcel details.</Alert>
      ) : (
        <Row>
          <Col>
            <ListGroup>
              <ListGroup.Item>
                <Label>Assessed Value: </Label>$
                {parcelDetail?.evaluations
                  ?.find(e => e.key === EvaluationKeys.Assessed)
                  ?.value?.toLocaleString('en-US', { minimumFractionDigits: 2 })}
              </ListGroup.Item>
            </ListGroup>
            <ListGroup>
              <ListGroup.Item>
                <div>
                  <Label>Address: </Label>
                  {parcelDetail?.address?.line1}
                </div>
                <div>
                  {parcelDetail?.address?.city}, {parcelDetail?.address?.province}{' '}
                  {parcelDetail?.address?.postal}
                </div>
              </ListGroup.Item>
              <ListGroup.Item>
                <Label>PID: </Label> {parcelDetail?.pid}
              </ListGroup.Item>
              {parcelDetail?.projectNumber && (
                <ListGroup.Item>
                  <Label>RAEG or SPP:</Label> {parcelDetail?.projectNumber}
                </ListGroup.Item>
              )}
            </ListGroup>
            <ListGroup>
              <ListGroup.Item>
                <Label>Agency: </Label>
                {parcelDetail?.agency}
              </ListGroup.Item>
              <ListGroup.Item>
                <Label>Classification: </Label>
                {parcelDetail?.classification}
              </ListGroup.Item>
            </ListGroup>
          </Col>
          <Col>
            {parcelDetail?.id &&
              !props?.disabled &&
              (!keycloak.hasAgency(parcelDetail?.agencyId) ? (
                <Link to={`/submitProperty/${parcelDetail?.id}?disabled=true`}>View</Link>
              ) : (
                <Link to={`/submitProperty/${parcelDetail?.id}`}>Update</Link>
              ))}
          </Col>
        </Row>
      )}
    </Container>
  );
};
