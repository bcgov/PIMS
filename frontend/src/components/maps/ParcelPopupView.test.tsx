import React from 'react';
import renderer from 'react-test-renderer';
import { ParcelPopupView } from './ParcelPopupView';
import { Router } from 'react-router-dom';
import { createMemoryHistory } from 'history';
import { render, cleanup } from '@testing-library/react';
import { useKeycloak } from '@react-keycloak/web';
import { IAddress, IParcel } from 'actions/parcelsActions';
import { Claims } from 'constants/claims';

const history = createMemoryHistory();

jest.mock('@react-keycloak/web');
(useKeycloak as jest.Mock).mockReturnValue({
  keycloak: {
    userInfo: {
      agencies: [1],
    },
    subject: 'test',
  },
});

const mockParcel = (agencyId: number, projectNumber?: string) => {
  var parcel: IParcel = {
    id: 1,
    pid: '1',
    pin: 1,
    latitude: 1,
    longitude: 1,
    projectNumber: projectNumber,
    classification: 'Test Classification',
    name: 'test name',
    description: 'Test Description',
    landArea: 100,
    classificationId: 1,
    zoning: '',
    agencyId: agencyId,
    isSensitive: false,
    landLegalDescription: 'Test Land Legal Description',
    address: ([
      {
        line1: '1234 Mock Street',
        administrativeArea: 'Test Administrative Area',
        provinceId: 'BC',
        postal: 'V0S1N0',
      },
    ] as unknown) as IAddress,
    evaluations: [],
    buildings: [],
    fiscals: [
      {
        fiscalYear: 2020,
        key: 'Key',
        value: 2,
      },
    ],
    zoningPotential: '',
  };
  return parcel;
};

describe('Parcel popup view', () => {
  afterEach(() => {
    cleanup();
  });
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <Router history={history}>
          <ParcelPopupView parcel={mockParcel(1)} />
        </Router>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('displays update option when user belongs to buildings agency', () => {
    const { getByText } = render(
      <Router history={history}>
        <ParcelPopupView parcel={mockParcel(1)} />
      </Router>,
    );
    expect(getByText(/Update/i)).toBeInTheDocument();
  });

  it('displays only view option when user not in properties agency', () => {
    const { getByText, queryByText } = render(
      <Router history={history}>
        <ParcelPopupView parcel={mockParcel(2)} />
      </Router>,
    );
    expect(getByText(/View/i)).toBeInTheDocument();
    expect(queryByText(/Update/i)).toBeNull();
  });

  it('always displays update option for sres', () => {
    (useKeycloak as jest.Mock).mockReturnValue({
      keycloak: {
        userInfo: {
          agencies: [1],
          roles: [Claims.ADMIN_PROPERTIES],
        },
        subject: 'test',
      },
    });
    const { getByText } = render(
      <Router history={history}>
        <ParcelPopupView parcel={mockParcel(2)} />
      </Router>,
    );
    expect(getByText(/Update/i)).toBeInTheDocument();
  });

  it('displays clickable project number when the property belongs to a project', () => {
    const { getByText } = render(
      <Router history={history}>
        <ParcelPopupView parcel={mockParcel(1, 'Test-Project-Number')} />
      </Router>,
    );
    const projectNum = getByText(/Test-Project-Number/);
    expect(getByText(/SPP/)).toBeInTheDocument();
    expect(projectNum).toBeInTheDocument();
  });

  it('does not display project number field when none present', () => {
    const { queryByText } = render(
      <Router history={history}>
        <ParcelPopupView parcel={mockParcel(1)} />
      </Router>,
    );
    expect(queryByText(/SPP/)).toBeNull();
  });
});
