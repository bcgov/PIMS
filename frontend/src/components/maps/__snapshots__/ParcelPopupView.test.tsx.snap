// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Parcel popup view renders correctly 1`] = `
<div
  className="parcelPopup container-fluid"
>
  <div
    className="row"
  >
    <div
      className="col"
    >
      <div
        className="list-group"
        onKeyDown={[Function]}
      >
        <div
          className="pid list-group-item"
          data-rb-event-key={null}
          disabled={false}
          onClick={[Function]}
        >
          1
        </div>
        <div
          className="name list-group-item"
          data-rb-event-key={null}
          disabled={false}
          onClick={[Function]}
        >
          test name
        </div>
        <div
          className="description list-group-item"
          data-rb-event-key={null}
          disabled={false}
          onClick={[Function]}
        >
          Test Description
        </div>
      </div>
      <div
        className="list-group"
        onKeyDown={[Function]}
      >
        <div
          className="list-group-item"
          data-rb-event-key={null}
          disabled={false}
          onClick={[Function]}
        >
          <p
            className="label undefined"
          >
            Assessed Value:
          </p>
          $
        </div>
      </div>
      <div
        className="list-group"
        onKeyDown={[Function]}
      >
        <div
          className="list-group-item"
          data-rb-event-key={null}
          disabled={false}
          onClick={[Function]}
        >
          <div />
          <div>
            , 
             
          </div>
        </div>
      </div>
      <div
        className="list-group"
        onKeyDown={[Function]}
      >
        <div
          className="list-group-item"
          data-rb-event-key={null}
          disabled={false}
          onClick={[Function]}
        >
          <p
            className="label undefined"
          >
            Agency:
          </p>
        </div>
        <div
          className="list-group-item"
          data-rb-event-key={null}
          disabled={false}
          onClick={[Function]}
        >
          <p
            className="label undefined"
          >
            Classification: 
          </p>
          Test Classification
        </div>
      </div>
    </div>
  </div>
  <div
    className="menu row"
  >
    <div
      className="col"
    >
      <a
        href="/mapview/1?disabled=true&loadDraft=false&sidebar=true"
        onClick={[Function]}
      >
        View
      </a>
      <a
        href="/mapview/1?disabled=false&loadDraft=false&sidebar=true"
        onClick={[Function]}
      >
        Update
      </a>
    </div>
  </div>
</div>
`;
